From 4f80c1dbd4168cbce0e470cfd5639684d5939693 Mon Sep 17 00:00:00 2001
From: Aurelien Jacquiot <a-jacquiot@ti.com>
Date: Fri, 10 Jun 2016 19:59:57 -0400
Subject: [PATCH 86/91] soc: ti: add queue flush helper macro

This patch adds a macro for the knav_queue_device_control
helper for KNAV_QUEUE_FLUSH.

Signed-off-by: Aurelien Jacquiot <a-jacquiot@ti.com>
Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
---
 include/linux/soc/ti/knav_helpers.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/soc/ti/knav_helpers.h b/include/linux/soc/ti/knav_helpers.h
index 42b7882..76db84b 100644
--- a/include/linux/soc/ti/knav_helpers.h
+++ b/include/linux/soc/ti/knav_helpers.h
@@ -106,4 +106,7 @@ static inline void knav_dma_get_words(u32 *words, int num_words, u32 *desc)
 #define knav_queue_get_count(q)	knav_queue_device_control(q, \
 				KNAV_QUEUE_GET_COUNT, (unsigned long)NULL)
 
+#define knav_queue_flush(q)	knav_queue_device_control(q,		\
+				KNAV_QUEUE_FLUSH, (unsigned long)NULL)
+
 #endif /* __SOC_TI_KEYSTONE_NAVIGATOR_HELPERS_H__ */
-- 
1.9.1

